### DISCUSSION TITLE
Trigger - access to action's actor

### DISCUSSION THREAD

**Discussion by KostarSf** [excaliburjs/Excalibur#3099] [Ideas]
Hi! It would be very useful to have access to the Actor, which calls Trigger's action callback. Something like this:

```ts
const trigger = new Trigger({
  // rest
  action: (actor) => {
    // do action
  }
})
```

I want to use triggers as areas of view for AI enemies, so it's important for me to know which Actor called action. Now I have to do my own implementation of the trigger, which takes this into account.

It would be great if it appeared in Excalibur!

**Comment by KostarSf** [comment]
It would also be very nice if the filter callback return type served as a typeguard and narrowed the actor type in the callback of the action.
Like this:

```ts
const trigger = new Trigger({
  // ...
  target: player, 
  // or
  filter: (actor) => actor instanceof Player
  action: (player) => {
    // type of player is Player here
  }
})
```

**Reply by eonarheim** [reply]
Oh yes, I agree!

**Reply by eonarheim** [reply]
I made an issue to this effect https://github.com/excaliburjs/Excalibur/issues/3157

**Reply by KostarSf** [reply]
I am very glad that you have implemented this feature, thank you very much! 🙏
